modules/kernel/src/org/apache/axis2/deployment/DescriptionBuilder.java

```diff
@@ -691,15 +698,43 @@ protected String getValue(String in) {
     }
 
     protected String replaceSystemProperty(String text) {
+        String[] sysRefs = StringUtils.substringsBetween(text, SYS_PROPERTY_PLACEHOLDER_PREFIX, PLACEHOLDER_SUFFIX);
+        String[] envRefs = StringUtils.substringsBetween(text, ENV_VAR_PLACEHOLDER_PREFIX, PLACEHOLDER_SUFFIX);
+
+        // Resolves system property references ($sys{ref}) in an individual string.
+        if (sysRefs != null) {
+            for (String ref : sysRefs) {
+                String property = System.getProperty(ref);
+                if (StringUtils.isNotEmpty(property)) {
+                    text = text.replaceAll(Pattern.quote(SYS_PROPERTY_PLACEHOLDER_PREFIX + ref + PLACEHOLDER_SUFFIX), property);
+                } else {
+                    throw new RuntimeException("Error while retrieving " + ref + " system property");
```

Comment:
Do we need to throw a runtime exception? Or is printing a warn log sufficient?
Also, the error message need to be modified to something like  'system property unavailable for" + ref

Both the comments are valid for the environment variable aswell

Reply:
Removed the runtime exception and updated the code to print a warn log in 51ff278a5422064f95aa588d469cd46c30f78ca7